### ISSUE TITLE
Integrate `_from_config` functions better with the rest of config

### ISSUE THREAD

**Issue by spladug** [reddit/baseplate.py#141]
It's hard to take advantage of more complex structuring with them, for example it's impossible to use nested keys:

```ini
thrift.service_a.endpoint = ...
thrift.service_b.endpoint = ...
```

wouldn't work with `thrift_pool_from_config` since it can't handle multi-level prefixes.

**Comment by spladug** [comment]
One option would be to build `baseplate.config.Parser` objects for each of these so they can be integrated into the application's main config spec all in one place. Then the above thrift config spec would look like:

```python
cfg = parse_config(app_config, {
    "thrift": config.DictOf(ThriftPool),
})

cfg.service_a.connect()
cfg.service_b.connect()
```

**Comment by spladug** [comment]
This is addressed in a345281020cb45fedfa3f78eadadecb8c09bbadd
